### DISCUSSION TITLE
Documentation

### DISCUSSION THREAD

**Discussion by GnomiBerlin** [khoih-prog/AsyncMQTT_Generic#13] [Q&A] [answered]
Sorry, found no other way.

Where are the parameters explained in your documentation?
e.g.  mqttClient.publish(PubTopic, 0, true, "ESP32 Test");

**Comment by khoih-prog** [comment, marked as answer]
At least read the `README.md` before posting

- [**Basic Usage**](https://github.com/khoih-prog/AsyncMQTT_Generic#basic-usage)
- [**uint16_t publish(....)**](https://github.com/khoih-prog/AsyncMQTT_Generic/blob/main/docs/2.-API-reference.md#uint16_t-publishconst-char-topic-uint8_t-qos-bool-retain-const-char-payload--nullptr-size_t-length--0-bool-dup--false-uint16_t-message_id--0)

**Comment by GnomiBerlin** [comment]
I read the README.md but missed this part: **Basic Usage
 --> Please check [API documentation]**

I just searched for e.g. mqttClient.publish()  which brought me just to the pasted example. Perhaps "Basic usage/API Reference" should be more at the beginning then in the middle of the long README.
Thank you.

**Comment by VladislavHolets** [comment]
Hello
I am sorry to disturb you, yet, I can't find a link for the library you are using for the ENC28j60 module in the documentation, and there is no one included in the platformio config.
Also, I wonder if there is a way of using this library with stm32f401 board( black pill ) and enc28j60 module (utilising SPI)?
Thank you!

**Comment by khoih-prog** [comment]
Async is not supporting `STM32F4` using `ENC28J60`. Possibly no hope in the far future.
Try **LAN8720** with STM32F4, using STM32 core v2.2.0

Only one board currently supporting **ENC28J60 Async** is **ESP8266**.

Check [**Currently supported Boards**](https://github.com/khoih-prog/AsyncMQTT_Generic#currently-supported-boards)
